backend/src/main/java/de/wirvsvirus/trackyourbed/CapacityService.java

```diff
@@ -0,0 +1,91 @@
+package de.wirvsvirus.trackyourbed;
+
+import de.wirvsvirus.trackyourbed.entity.Bed;
+import de.wirvsvirus.trackyourbed.entity.Capacity;
+import de.wirvsvirus.trackyourbed.entity.Department;
+import de.wirvsvirus.trackyourbed.entity.DepartmentCapacity;
+import de.wirvsvirus.trackyourbed.entity.Hospital;
+import de.wirvsvirus.trackyourbed.entity.HospitalCapacity;
+import de.wirvsvirus.trackyourbed.entity.Ward;
+import de.wirvsvirus.trackyourbed.entity.WardCapacity;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CapacityService {
+
+  private static final String NORMAL = "normal";
+  private static final String IMC = "imc";
+  private static final String ICU = "icu";
+  private static final String COVID = "covid";
+  private static final String COVID_ICU = "covid-icu";
+
+  public HospitalCapacity calculateHospitalCapacity(final Hospital hospital) {
+    final List<DepartmentCapacity> departmentCapacities = hospital.getDepartments().stream()
+        .map(this::calculateDepartmentCapacity)
+        .collect(Collectors.toList());
+    final HospitalCapacity hospitalCapacity = new HospitalCapacity(hospital.getName());
+    hospitalCapacity.add(departmentCapacities);
+    hospitalCapacity.setDepartmentCapacities(departmentCapacities);
```

Comment:
So the list contains both the information for _maximum capacity_ and _available_ beds? That works, but the name `.add()` is a bit misleading. A better name could be `.setFreeCapacity()`.